components/script/script_thread.rs

```diff
@@ -2309,69 +2310,70 @@ impl ScriptThread {
-                                match strval {
-                                    Ok(ConversionResult::Success(s)) => {
-                                        Some(JsEvalResult::Ok(String::from(s).as_bytes().to_vec()))
-                                    },
-                                    _ => None,
-                                }
-                            }
-                        } else {
-                            Some(JsEvalResult::NoContent)
-                        };
-                    }
-                }
-
-                if is_javascript {
-                    load_data.url = ServoUrl::parse("about:blank").unwrap();
-                }
-
                 self.script_sender
                     .send((parent_pipeline_id, ScriptMsg::LoadUrl(load_data, replace)))
                     .unwrap();
             }
         }
     }
 
+    fn eval_js_url(&self, pipeline_id: PipelineId, load_data: &mut LoadData) {
+        {
+            // Turn javascript: URL into JS code to eval, according to the steps in
+            // https://html.spec.whatwg.org/multipage/#javascript-protocol
+
+            // This slice of the URL’s serialization is equivalent to (5.) to (7.):
+            // Start with the scheme data of the parsed URL;
+            // append question mark and query component, if any;
+            // append number sign and fragment component if any.
+            let encoded = &load_data.url[Position::BeforePath..];
+
+            // Percent-decode (8.) and UTF-8 decode (9.)
+            let script_source = percent_decode(encoded.as_bytes()).decode_utf8_lossy();
+
+            // Script source is ready to be evaluated (11.)
+            let window = self.documents.borrow().find_window(pipeline_id);
+
+            if let Some(window) = window {
+                let _ac = JSAutoCompartment::new(self.get_cx(), window.reflector().get_jsobject().get());
+                rooted!(in(self.get_cx()) let mut jsval = UndefinedValue());
+                window.upcast::<GlobalScope>().evaluate_js_on_global_with_result(
+                    &script_source, jsval.handle_mut());
+
+                load_data.js_eval_result = if jsval.get().is_string() {
+                    unsafe {
+                        let strval = DOMString::from_jsval(self.get_cx(),
+                                                           jsval.handle(),
+                                                           StringificationBehavior::Empty);
+                        match strval {
+                            Ok(ConversionResult::Success(s)) => {
+                                Some(JsEvalResult::Ok(String::from(s).as_bytes().to_vec()))
+                            },
+                            _ => None,
+                        }
+                    }
+                } else {
+                    Some(JsEvalResult::NoContent)
+                };
+            }
+        };
```

Comment:
I don't think this surrounding {} is necessary any more.

Reply:
Removed the block in https://github.com/servo/servo/pull/17083/commits/709cd3a0b636f601c0a4ca606c241500eca13789?w=1, does that look ok?